src/docker-osx-dev

```diff
@@ -375,17 +381,74 @@ function init_boot2docker {
     log_info "Starting Boot2Docker VM"
     boot2docker start --vbox-share=disable
   fi
+
+  configure_boot2docker
+}
+
+################################################################################
+# docker-machine manipulation
+################################################################################
+
+#
+# Read a json string from stdin and return the value of the key provided as first argument
+# Uses head -n 1 to avoid duplicates
+#
+function get_json_value () {
+    sed -n 's/.*"'"$*"'": *"\([^"]*\)".*/\1/p' | head -n 1
+}
+
+#
+# Configures variables based on a the docker-machine inspect from $DOCKER_MACHINE_NAME
+#
+function configure_docker_machine {
+  DOCKER_HOST_NAME=$DOCKER_MACHINE_NAME
+  DOCKER_MACHINE_INSPECT=$(docker-machine inspect $DOCKER_MACHINE_NAME)
+  DOCKER_HOST_USER=$(echo -e "$DOCKER_MACHINE_INSPECT" | get_json_value "SSHUser")
+  DOCKER_HOST_IP=$(echo -e "$DOCKER_MACHINE_INSPECT" | get_json_value "IPAddress")
+  DOCKER_MACHINE_STORE_PATH=$(echo -e "$DOCKER_MACHINE_INSPECT" | get_json_value "StorePath")
+
+  DOCKER_HOST_SSH_URL="$DOCKER_HOST_USER@$DOCKER_HOST_IP"
+  DOCKER_HOST_SSH_KEY="$DOCKER_MACHINE_STORE_PATH/id_rsa"
+  DOCKER_HOST_SSH_COMMAND="docker-machine ssh $DOCKER_MACHINE_NAME"
+  log_debug "Running for the docker-machine name: $DOCKER_MACHINE_NAME"
+}
+
+#
+# Initializes and starts up the docker-machine VM.
+#
+function init_docker_machine {
+  log_info "Initializing docker machine $DOCKER_MACHINE_NAME"
+  docker-machine start "$DOCKER_MACHINE_NAME"
+  eval $(docker-machine env "$DOCKER_MACHINE_NAME")
+  configure_docker_machine
+}
```

Comment:
So, here, this eval will run something like this:

```
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.103:2376"
export DOCKER_CERT_PATH="/Users/someone/.docker/machine/machines/local"
export DOCKER_MACHINE_NAME="local"
```

We can see that `DOCKER_MACHINE_NAME` will be redefined.

Also, as in the `docker-machine` documentation, if the user runs `eval "$(docker-machine env <machine-name>)"`, it will define the `DOCKER_MACHINE_NAME` to the env. Then if he runs `docker-osx-dev`, he would not need to pass the `--machine-name` argument.

I guess this is fine, I can add to the `Readme.md` about this behavior.